src/transformStateWithClones.js

```diff
@@ -65,7 +65,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const newStates = [];
+  const newState = { ...state };
+
+  transforms.forEach(move => {
+    switch (move.operation) {
+      case 'addProperties':
+        Object.assign(newState, move.properties);
+        newStates.push({ ...newState });
```

Comment:
newStates.push({ ...newState }) can be written only once after switch case construction